Add Login Page

jaminko opened the pull request.

Commit:
Add Login Without Password Test

Commit:
Add Login Without User Name Test

Commit:
Add Login With Empty Fields Test

varajan committed:
Enable action: build&run tests

varajan committed:
Edit action: run on Windows instead of Ubuntu

varajan committed:
Fix Login button id: login -> loginBtn

Commit:
Add Login Field Signature Test

Commit:
Edit expected result for 'LoginFieldSignatureTest' (according to requirements)

Login filed should be labeled 'User' but labeled 'Login'

Commit:
Change expected signature results according to requirements in README file

Commit:
Add Valid Login Test

Commit:
Merge branch 'LoginPage' of https://github.com/jaminko/web.test into LoginPage

varajan committed:
#3 Fix 'User' label

varajan committed:
#2 Fix 'Password' label

varajan commented on DepositeCalcTests/Tests/LoginPageTests.cs:
```diff
@@ -0,0 +1,134 @@
+            loginBtn.Click();
+
+            // Assert
+            Assert.IsTrue(driver.FindElement(By.Id("amount")).Displayed);
+            Assert.AreNotEqual(currentURL, driver.Url);
```
I think it is better to verify the current URL is equal to '.../Calculator'

jaminko replied:
Got it!
In my opinion, we should also update the README.MD file:
row 10 -  change * Login filed, labeled 'User' to  * Login filed, labeled 'User:'
row 11 -  change * Password filed, labeled 'Password'change to  * Password filed, labeled 'Password:'

Commit:
Update locators after fixing

 - change locators for 'Login' CTA button;
- change assertion for Valid Login Test.

Commit:
Merge 3 tests to 1

Commit:
Change locators for login and password fields

Commit:
Implement Page Object Model

Commit:
Create method Login

Commit:
Add Explicit Wait

Commit:
Downgrade chrome driver to version 106

varajan commented on DepositeCalcTests/Tests/LoginPageTests.cs:
```diff
@@ -0,0 +1,60 @@
+        {
+            ChromeOptions options = new ChromeOptions { AcceptInsecureCertificates = true };
+            driver = new ChromeDriver(options);
+            driver.Manage().Timeouts().PageLoad = TimeSpan.FromSeconds(30);
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(5);
+            driver.Url = "https://localhost:5001/";
```
фівадфіва жо жаофівжао 
```suggestion
            driver.Manage().Timeouts().PageLoad = TimeSpan.FromSeconds(40);
            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(65);
            driver.Url = "https://localhost:5001/";
```

varajan commented on DepositeCalcTests/Tests/LoginPageTests.cs:
```diff
@@ -0,0 +1,60 @@
+            var loginPage = new LoginPage(driver);
+
+            // Act
+            loginPage.Login(login, password);
```
укепвфуі емуцк ек

Commit:
Changed error msg text (Incorrect credentials)

varajan committed:
Fix 'Incorrect credentials' bug

Commit:
Merge branch 'LoginPage' of https://github.com/jaminko/web.test into LoginPage
